docs/sketches/interaction-sketch.md

```diff
@@ -0,0 +1,18 @@
+# Interaction Sketch
+
+- Don't discriminate authenticated user (both students and teachers can log in).
+
+## Protecting Classroom Invites
+
+#### How to associate a CodeGarten account with an Institutional ID
+- The user inserts his institutional e-mail and ID and will receive a verification link on it, which will take him back to CodeGarten, verifying the ownership of the institutional e-mail
+
+#### How do Classroom Invites work
+- Teacher generates an invite link for the classroom
+- Students, via that link, will have to log in with their CodeGarten account (create one if it doesn't exist), and link their institutional e-mail with their CodeGarten account if needed (process described above)
```

Comment:
I would avoid having to manage user's credentials. Instead, we can delegate authentication to GitHub.